libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/UpcomingRoadObject.kt

```diff
@@ -1,25 +1,34 @@
 package com.mapbox.navigation.base.trip.model.roadobject
 
+import com.mapbox.navigation.base.trip.model.roadobject.distanceinfo.RoadObjectDistanceInfo
+
 /**
- * Holds the road objects and the distance to the point where the alert occurs,
- * or start an alert if it has length.
+ * Holds the road objects and the distance to the start of the object.
+ * There are two sources of road objects: active route and the electronic horizon.
+ * Road objects coming from the electronic horizon will also have [RoadObjectDistanceInfo]
+ * Available [RoadObjectDistanceInfo] types are:
+ * - [RoadObjectDistanceInfoType.GANTRY]
+ * - [RoadObjectDistanceInfoType.LINE]
+ * - [RoadObjectDistanceInfoType.POINT]
+ * - [RoadObjectDistanceInfoType.POLYGON]
+ * - [RoadObjectDistanceInfoType.SUB_GRAPH]
  *
  * @param roadObject road object
- * @param distanceToStart distance to the start of the alert.
- * If the object has a length, and we've passed the start point,
- * **this value will be negative** until we cross the finish point of the objects's geometry.
- * This negative value, together with [RoadObjectGeometry.length]
- * can be used to calculate the distance since the start of an object.
+ * @param distanceToStart remaining distance to the start of the object.
+ * @param distanceInfo provides extra distance details for the road objects. It will be non-null
+ * for objects coming from the electronic horizon and null for objects that are on the current route
+ * that we are actively navigating on.
  */
 class UpcomingRoadObject private constructor(
```

Comment:
Migrate to `internal` constructor and internal factory.

Reply:
done